### PR TITLE
Refactor pipeline and review build cache

### PR DESCRIPTION
**Pull request by hfgbarrigas**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [x] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: .github/actions/init_rust_job/action.yml [hunk 1]
```diff
@@ -2,42 +2,46 @@ name: CLI rust job init
...
       - macos
       - linux
-  assets_version:
-    description: Version of the assets to fetch from CDN
-    required: true
 
 runs:
   using: 'composite'
   steps:
-    - name: move CLI to root
```
Comment (yoav-lavi) [inline .github/actions/init_rust_job/action.yml:23]: Why was this removed?
Some of the tests / tooling rely on this

Comment (hfgbarrigas) [reply .github/actions/init_rust_job/action.yml:23]: no longer needed

**Review by yoav-lavi — APPROVED** [review]
Nice work!
